dask/array/core.py

```diff
@@ -531,43 +569,73 @@ def dask_slice(out_name, in_name, shape, blockdims, indexes,
     Example
     -------
 
-    >>> dask_slice('y', 'x', (100,), [(20, 20, 20, 20, 20)], (slice(10, 35),))  #  doctest: +SKIP
+    >>> fancy_slice('y', 'x', (100,), [(20, 20, 20, 20, 20)], (slice(10, 35),))  #  doctest: +SKIP
     {('y', 0): (getitem, ('x', 0), (slice(10, 20),)),
      ('y', 1): (getitem, ('x', 1), (slice( 0, 15),))}
+
+    See Also
+    --------
+
+    take - handle slicing with lists ("fancy" indexing)
+    dask_slice - handle slicing with slices and integers
+    """
+    where_list = [i for i, ind in enumerate(indexes) if isinstance(ind, list)]
```

Comment:
We can do both.  I was just at about my limit for complexity while I was building this and didn't want to think about other cases.  Both of those sound good though.